src/log/log_slot.c

```diff
@@ -262,7 +262,7 @@ __wt_log_slot_new(WT_SESSION_IMPL *session)
 		/*
 		 * If we didn't find any free slots signal the worker thread.
 		 */
-		(void)__wt_cond_auto_signal(session, conn->log_wrlsn_cond);
+		WT_RET(__wt_cond_auto_signal(session, conn->log_wrlsn_cond));
```

Comment:
I modeled the return after the failure line above it, a call to `__wt_log_acquire` (which can fail for a wide variety of reasons, including malloc failures). If this is dangerous, I suspect that return is problematic as well.

That said, my plan is to change `__wt_cond_auto_signal` to not return anything as part of [WT-2822](https://jira.mongodb.org/browse/WT-2822), so this specific return shouldn't be around too much longer.